pkg/network/discovery.go

```diff
@@ -171,15 +178,24 @@ func (d *DefaultDiscovery) tryAddress(addr string) {
 	}
 }
 
+// Close stops discoverer pool processing making discoverer almost useless.
+func (d *DefaultDiscovery) Close() {
+	close(d.requestCh)
```

Comment:
I think, there is a race condition and a possible panic (`send on a closed channel`).
```golang
go d.RequestRemote(1) // blocks on send
d.Close() // makes previous line panic
```
Maybe smth like this will be better
1.  `isDead = true` (ensure no new sends will occur)
2. `drain requestCh` (ensure all blocked sends are processed)
3. `close(requestCh)`

Reply:
Fixed.